.lgtm/cpp-queries/missing-argument-null-check.c

```diff
@@ -0,0 +1,29 @@
+#include <assert.h>
+
+typedef struct {
+    char *string;
+} StructWithString;
+
+void bad_deref(StructWithString *data)
+{
+    char *string = data->string;
+}
+
+void good_with_assert(StructWithString *data)
+{
+    assert(data != NULL);
+    char *string = data->string;
```

Comment:
I think we should also have `good_with_if` because in many cases that's the best thing to do.

Reply:
Good point.